src/gui/mzroll/mzfileio.cpp

```diff
@@ -223,112 +223,149 @@ PK$PEAK: m/z int. rel.int.
-			   else if (mw) { cpd->mass=mw; cpd->precursorMz=precursor; }
-               //cpd->mass=mw;
-               //cpd->precursorMz=mw;
-               cpd->db=dbname;
-               cpd->fragment_mzs = mzs;
-               cpd->fragment_intensity = intest;
-			   cpd->expectedRt=retentionTime;
-               DB.addCompound(cpd);
-               compoundCount++;
+        if (line.startsWith("MW:", Qt::CaseInsensitive)) {
+            currentCompound->mass = line.mid(3, line.length())
+                                        .simplified()
+                                        .toDouble();
+        } else if (line.startsWith("CE:", Qt::CaseInsensitive)) {
+            currentCompound->collisionEnergy = line.mid(3, line.length())
+                                                   .simplified()
+                                                   .toDouble();
+        } else if (line.startsWith("ID:", Qt::CaseInsensitive)) {
+            QString id = line.mid(3, line.length()).simplified();
+            if (!id.isEmpty())
+                currentCompound->id = id.toStdString();
+        } else if (line.startsWith("LOGP:", Qt::CaseInsensitive)) {
+            currentCompound->logP = line.mid(5, line.length())
+                                        .simplified()
+                                        .toDouble();
+        } else if (line.startsWith("RT:", Qt::CaseInsensitive)) {
+            currentCompound->expectedRt = line.mid(3, line.length())
+                                              .simplified()
+                                              .toDouble();
+        } else if (line.startsWith("SMILE:", Qt::CaseInsensitive)) {
+            QString smileString = line.mid(7, line.length()).simplified();
+            if (!smileString.isEmpty())
+                currentCompound->smileString = smileString.toStdString();
+        } else if (line.startsWith("SMILES:", Qt::CaseInsensitive)) {
+            QString smileString = line.mid(8, line.length()).simplified();
+            if (!smileString.isEmpty())
+                currentCompound->smileString = smileString.toStdString();
+        } else if (line.startsWith("PRECURSORMZ:", Qt::CaseInsensitive)) {
+            currentCompound->precursorMz = line.mid(13, line.length())
+                                               .simplified()
+                                               .toDouble();
+        } else if (line.startsWith("EXACTMASS:", Qt::CaseInsensitive)) {
+            currentCompound->mass = line.mid(10, line.length())
+                                        .simplified()
+                                        .toDouble();
+        } else if (line.startsWith("ADDUCT:", Qt::CaseInsensitive)) {
+            currentCompound->adductString = line.mid(8, line.length())
+                                                .simplified()
+                                                .toStdString();
+        } else if (line.startsWith("FORMULA:", Qt::CaseInsensitive)) {
```

Comment:
Check if formulae in the form of ".H2O" or "(H2O)" are handled somewhere. If not, then remove '.' and '()' from the QString as well.